R/onLoad.R

```diff
@@ -2,6 +2,31 @@
   register_upgrade_message("shiny", "1.6.0")
   register_upgrade_message("rmarkdown", "2.7")
   .dependency_cache <<- cachem::cache_mem(max_age = 1)
+
+  # Call setup_input_handlers() when both bslib and shiny are loaded
+  if (!setup_input_handlers()) {
+    # shiny is not yet loaded, if it ever gets loaded we'll register then
+    setHook(packageEvent("shiny", "onLoad"), function(...) {
+      setup_input_handlers()
+    })
+  }
 }
 
 .dependency_cache <- NULL
+
+# Returns TRUE if input handlers were successfully loaded
+setup_input_handlers <- function() {
+  if (!"shiny" %in% loadedNamespaces()) {
+    return(FALSE)
+  }
+
+  if (!"registerInputHandler" %in% names(asNamespace("shiny"))) {
+    # In devtools::load_all(), the shiny namespace is loaded but empty at the
+    # time bslib loads. It's OK not to do the registration now, it'll just
+    # happen when shiny finishes loading.
+    return(FALSE)
+  }
+
+  shiny::registerInputHandler("bslib.taskbutton", input_task_button_input_handler)
```

Comment:
I'm getting an error when repeatedly calling `load_all()`?

```suggestion
  shiny::registerInputHandler("bslib.taskbutton", input_task_button_input_handler, force = TRUE)
```

Reply:
Hmmm, do we want to fix that? The error serves a purpose if you're not using `load_all`.

OTOH, in this case, `bslib.taskbutton` is unlikely to collide with anything and I'm assuming you and @gadenbuie probably use `load_all` a lot...?